packages/contracts/src/dollar/libraries/LibUbiquityPool.sol

```diff
@@ -212,243 +444,297 @@ library LibUbiquityPool {
      * @dev 1. `redeemDollar()`
      * @dev 2. `collectRedemption()`
      * @dev This is done in order to prevent someone using a flash loan of a collateral token to mint, redeem, and collect in a single transaction/block
-     * @param collateralAddress address of the collateral token being collected
+     * @param collateralIndex Collateral token index being collected
+     * @return collateralAmount Amount of collateral tokens redeemed
      */
-    function collectRedemption(address collateralAddress) internal {
+    function collectRedemption(
+        uint256 collateralIndex
+    ) internal returns (uint256 collateralAmount) {
         UbiquityPoolStorage storage poolStorage = ubiquityPoolStorage();
+
         require(
-            poolStorage.lastRedeemed[msg.sender] +
-                poolStorage.redemptionDelay >=
-                block.number,
-            "Too soon to collect"
+            poolStorage.redeemPaused[collateralIndex] == false,
+            "Redeeming is paused"
+        );
+        require(
+            (
+                poolStorage.lastRedeemed[msg.sender].add(
+                    poolStorage.redemptionDelay
+                )
+            ) <= block.number,
+            "Too soon"
```

Comment:
I would make it more explicit "Too soon" can mean anything. Perhaps "Too soon to collect redemption" ? .

Reply:
fixed https://github.com/ubiquity/ubiquity-dollar/pull/832/commits/39917d84af42b8fc5866ee02c63653170e5dc9f0